pages/create-release.js

```diff
@@ -0,0 +1,155 @@
+import { useState, useEffect } from 'react'
+import { supabase } from '@/utils/supabase'
+import { useUser } from '@/utils/context/user'
+
+const trueFalse = [
+  { id: 1, text: 'True', value: true },
+  { id: 2, text: 'False', value: false }
+]
+
+const releaseTypes = [
+  { id: 1, text: 'LP' },
+  { id: 2, text: 'EP' },
+  { id: 3, text: 'Single' },
+  { id: 4, text: 'Compilation' },
+  { id: 5, text: 'Soundtrack' },
+  { id: 6, text: 'Choose release type', isDisabled: true }
+]
+
+export default function CreateRelease() {
+  const { activeUser } = useUser()
+  const [newRelease, setNewRelease] = useState({
+    title: '',
+    artist: '',
+    label: '',
+    artwork: '',
+    downloadUrl: '',
+    pagePassword: '',
+    isPasswordProtected: trueFalse[1].value,
+    type: releaseTypes[5].text,
+    userId: ''
+  })
+
+  useEffect(() => {
+    if (activeUser) {
+      if (activeUser.type === 'Label') {
+        setNewRelease({
+          ...newRelease,
+          label: activeUser.name,
+          userId: activeUser.id
+        })
+      }
+      if (activeUser.type === 'Artist') {
+        setNewRelease({
+          ...newRelease,
+          artist: activeUser.name,
+          userId: activeUser.id
+        })
+      }
+    }
+  }, [activeUser])
+
+  const handleChange = (e) => {
+    setNewRelease({ ...newRelease, [e.target.id]: e.target.value })
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    const { data, error } = await supabase.from('releases').insert([newRelease])
+  }
+
+  return (
+    activeUser && (
```

Comment:
If _not_ an `activeUser`, then what? Is this to prevent a logged out user from seeing content on `/create-release`? What do they see?

If you're going to do this, it might be better to just return your template normally, then above that `return` do something like:

```jsx
if (!activeUser) {
  return <LoginForm />
}
```

Reply:
That is a good call. I was just trying to get the create release function working, still a lot more to do with error handling and what not.  Maybe instead of just sending the login form, it displays also displays message saying that they need to be a member to create a release, and why it's awesome to be a member?